runtime/interface.go

```diff
@@ -41,55 +43,130 @@ type Block struct {
+	UnsuspendAccount(address Address, caller Location) error
+	// returns true if account is suspended
+	IsAccountSuspended(address Address) (isSuspended bool, err error)
+}
+
+// AccountContracts manages contracts stored under accounts
+type AccountContracts interface {
+	// AccountContractCode returns the code associated with an account contract.
+	ContractCode(address AddressLocation) (code []byte, err error)
+	// UpdateAccountContractCode updates the code associated with an account contract.
+	UpdateContractCode(address AddressLocation, code []byte, caller Location) (err error)
+	// RemoveContractCode removes the code associated with an account contract.
+	RemoveContractCode(address AddressLocation, caller Location) (err error)
+	// Contracts returns a list of contract names under this account
+	Contracts(address AddressLocation, caller Location) (name []string, err error)
+}
 
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
+// AccountStorage stores and retrives account key/values
+type AccountStorage interface {
+	// Value gets a value for the given key in the storage, owned by the given account.
+	Value(key StorageKey, caller Location) (value StorageValue, err error)
+	// SetValue sets a value for the given key in the storage, owned by the given account.
+	SetValue(key StorageKey, value StorageValue, caller Location) (err error)
+	// ValueExists returns true if the given key exists in the storage, owned by the given account.
+	ValueExists(key StorageKey, caller Location) (exists bool, err error)
+	// StoredKeys returns list of keys and their sizes owned by this account
+	StoredKeys(address Address, caller Location) (iter StorageKeyIterator, err error)
+	// StorageUsed gets storage used in bytes by the address at the moment of the function call.
+	StorageUsed(address Address, caller Location) (value uint64, err error)
+	// Note: StorageCapacity has been moved to injected methods (similar to get balance)
+}
+
+// AccountKeys manages account keys
+type AccountKeys interface {
+	// AddAccountKey appends a key to an account.
+	AddAccountKey(address Address, publicKey []byte, caller Location) error
+	// RemoveAccountKey removes a key from an account by index.
+	RevokeAccountKey(address Address, index int, caller Location) (publicKey []byte, err error)
+	// AccountPublicKey returns the account key for the given index
+	AccountPublicKey(address Address, index int, caller Location) (publicKey []byte, err error)
+}
+
+type LocationResolver interface {
 	// GetCode returns the code at a given location
 	GetCode(location Location) ([]byte, error)
+	// ResolveLocation resolves an import location.
+	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
+}
+
+// Results are responsible to capture artifacts generated
+// when running the runnable
```

Comment:
"Runnable" isn't mentioned anywhere

```suggestion
// Results is an interface that is used to capture artifacts generated by executing a Cadence program. 
```